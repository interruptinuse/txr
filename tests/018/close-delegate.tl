(load "../common")

(defstruct refcount-close stream-wrap
  stream
  (count 1)

  (:method close (me throw-on-error-p)
    (put-line `close called, count @{me.count}`)
    (when (plusp me.count)
      (if (zerop (dec me.count))
        (close-stream me.stream throw-on-error-p)))))

(flow
  (with-stream (s (make-struct-delegate-stream
                    (new refcount-close
                         count 2
                         stream (open-file *load-path*))))
    (get-lines s))
  len
  prinl)
