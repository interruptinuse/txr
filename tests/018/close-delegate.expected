close called, count 2
close called, count 1
40
close called, count 2
close called, count 1
40
